File: solutions/r/raindrops/2/raindrops.R
raindrops <- function(number) {
	factors <- function (n) {
		#print(c("n=", n))
		if (n == 1) {return (c(1))}
		for (d in 1:(n %/% 2)) {
			#print ("d=",d)
			if (d == 1) {
				f <- c(1) 
			} else if  (n %% d == 0)
			{
				f <- c(f, d)
				#print(c("f=",f))
			}			
		}
		f <- c(f,n)
		return (f)
	} 
	
	rd <- ""
	for (f in factors(number)) {
		if (f == 3) { rd <- paste(rd,"Pling", sep="")}
		if (f == 5) { rd <- paste(rd,"Plang", sep="")}
		if (f == 7) { rd <- paste(rd,"Plong", sep="")}
	}
	if (rd == "") { rd <- as.character(number) }
	return(rd)

}

File: solutions/go/raindrops/2/raindrops.go
package raindrops

import "strconv"

// Convert - changes a number into raindrops
func Convert(n int) string {
    var rd string 
    if n % 3 == 0 { rd += "Pling"}
    if n % 5 == 0 { rd += "Plang"}
    if n % 7 == 0 { rd += "Plong"}
    if rd == "" { rd = strconv.Itoa(n)}
    return rd
}
